container/sagemaker/python_service.py

```diff
@@ -168,29 +172,56 @@ def on_post(self, req, res):
                           .decode('utf-8'))
         model_name = data['model_name']
         base_path = data['url']
-        try:
-            msg = self.grpc_client.add_model(model_name, base_path)
-            res.body = msg
-            res.status = falcon.HTTP_200
-        except Exception as e:  # pylint: disable=W0703
-            e = eval(str(e))  # pylint: disable=W0123
-            if e[0] == 409:
-                res.status = falcon.HTTP_409
-            else:
-                res.status = falcon.HTTP_500
-            res.body = e[1].encode('utf-8')
+        if self.validate_model_dir(base_path):
+            try:
+                msg = self.grpc_client.add_model(model_name, base_path)
+
+                # make sure all versions of model is ready before returning the call
+                with self._timeout(seconds=60):
+                    while True:
+                        model_status = self._get_model_status(model_name)
+                        if self._check_all_versions_available(model_status):
+                            break
+                        sleep(1)
+
+                res.body = msg
+                res.status = falcon.HTTP_200
+            except Exception as e:  # pylint: disable=W0703
+                e = eval(str(e))  # pylint: disable=W0123
```

Comment:
this is dangerous. never use eval on untrusted data. in fact, just never use eval.

Reply:
Removed eval().

proxy_client will throw http exceptions and python service will catch http exceptions and other exceptions separately.